src/tbb/arena.cpp

```diff
@@ -187,6 +187,16 @@ arena::arena ( market& m, unsigned num_slots, unsigned num_reserved_slots, unsig
         d1::task_group_context{ d1::task_group_context::isolated, d1::task_group_context::fp_settings };
     // Construct slots. Mark internal synchronization elements for the tools.
     task_dispatcher* base_td_pointer = reinterpret_cast<task_dispatcher*>(my_slots + my_num_slots);
+
+    if (static_cast<int>(num_slots) == tbb::info::default_concurrency()) {
+        const auto& numa_nodes_size = governor::get_numa_nodes_size();
+        int right = 0;
+        for (std::size_t i = 0; i < numa_nodes_size.size(); ++i) {
+            numa_intervals[i] = std::make_pair(right, right + numa_nodes_size[i]);
+            right += numa_nodes_size[i];
+        }
+    }
```

Comment:
```suggestion
    if (static_cast<int>(num_slots) == tbb::info::default_concurrency()) {
        for (auto& numa_index: tbb::info::numa_nodes()) {
            numa_intervals[numa_index] = std::make_pair(right, right + governor::get_numa_cores_count(numa_index));
        }
    }
```
Consider the following semantic